### ISSUE TITLE
runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by wemod123** [openfga/openfga#571]
Version: 0.3.7

When call the `../list-objects` API, get below errors in console
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xd54520]
goroutine 161 [running]:
github.com/openfga/openfga/pkg/typesystem.RewriteContainsSelf.func1(0x30?)
/home/runner/work/openfga/openfga/pkg/typesystem/typesystem.go:909
github.com/openfga/openfga/pkg/typesystem.WalkUsersetRewrite(0x0, 0x11127a8)
/home/runner/work/openfga/openfga/pkg/typesystem/typesystem.go:1101 +0x2d
github.com/openfga/openfga/pkg/typesystem.RewriteContainsSelf(0xc00071ba60?)
/home/runner/work/openfga/openfga/pkg/typesystem/typesystem.go:908 +0x25
github.com/openfga/openfga/pkg/typesystem.(*TypeSystem).IsDirectlyAssignable(...)
/home/runner/work/openfga/openfga/pkg/typesystem/typesystem.go:901
github.com/openfga/openfga/internal/graph.(*ConnectedObjectGraph).findIngressesWithRewrite(0xc00071ba60, 0xc00071a530, 0xc00071bd38, 0xc0006ce140?, 0xf?)
/home/runner/work/openfga/openfga/internal/graph/graph.go:202 +0xdad
github.com/openfga/openfga/internal/graph.(*ConnectedObjectGraph).findIngressesWithRewrite(0xc00071ba60, 0xc00071a530, 0xc00071bd38, 0xc0006170f8?, 0x5?)
/home/runner/work/openfga/openfga/internal/graph/graph.go:233 +0xbaa
...
...
...
/home/runner/work/openfga/openfga/internal/graph/graph.go:129 +0x1cf
github.com/openfga/openfga/internal/graph.(*ConnectedObjectGraph).RelationshipIngresses(...)
/home/runner/work/openfga/openfga/internal/graph/graph.go:113
github.com/openfga/openfga/pkg/server/commands.(*ConnectedObjectsCommand).streamedConnectedObjects(0xc00061cc00, {0x1245370, 0xc000795d80}, 0xc000516780, 0xc0000a4240, 0xc000792450, 0xc00061736c)
/home/runner/work/openfga/openfga/pkg/server/commands/connected_objects.go:158 +0xcb7
github.com/openfga/openfga/pkg/server/commands.(*ConnectedObjectsCommand).reverseExpandTupleToUserset.func1()
/home/runner/work/openfga/openfga/pkg/server/commands/connected_objects.go:390 +0x1e6
golang.org/x/sync/errgroup.(*Group).Go.func1()
/home/runner/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
/home/runner/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
```

Seems just simply assigns the value of pointer with nil.

This fatal error will cause downtime and restart server.

**Comment by rhamzeh** [comment]
Thanks for your report @wemod123 - the team will take a look at this shortly!

**Comment by adriantam** [comment]
Thank you @wemod123 for reporting this. 

We are able to reproduce the issue with the following model
```

model
  schema 1.1
type user
type folder
  relations
    define owner: [user]
    define bad: [user,user:*] or owner
type document
  relations
    define can_read: bad from parent
    define parent: [document,folder]
    define viewer: [user,user:*]
    ```
    
    and the list object request with 
    ```
    {
    "type": "document",
    "relation": "can_read",
    "user": "user:anne",
    "contextual_tuples": {
        "tuple_keys": [
            {
                "object": "folder:a",
                "relation": "owner",
                "user": "user:anne"
            },
            {
                "object": "document:c",
                "relation": "parent",
                "user": "folder:a"
            },
            {
                "object": "document:d",
                "relation": "parent",
                "user": "document:c"
            }
        ]
    }
}
```

**Comment by adriantam** [comment]
Similar issue but has different behaviour.  With the following model

```
model
  schema 1.1
type user
type folder
  relations
    define owner: [user]
    define viewer: [user,user:*] or owner
type document
  relations
    define can_read: viewer from parent
    define parent: [document,folder]
    define viewer: [user,user:*]
```

The following list object request will return `{document:c, document:d}`.  `document:d` should not returned as there is no relation `document:d#can_read@user:anne`.

**Comment by wemod123** [comment]
@adriantam I have a very similar model as you showed with parent relations + user#group etc. aiming to list objects with inherit relations including user groups, good to see you reproduce and address the issue quickly.

**Comment by miparnisari** [comment]
@wemod123 this bug is fixed in v0.4.0 which will be released shortly. Thanks for your report! 🙇‍♀️

**Comment by aaguiarz** [comment]
Hey @wemod123 

Were you able to finally use OpenFGA? Would you mind emailing me at andres.aguiar at openfga.dev and share what has been your experience with it?

Thanks
